src/helpers/DataRequirementHelpers.ts

```diff
@@ -391,3 +391,57 @@ export function codeLookup(dataType: string, attribute: string): string | null {
   }
   return null;
 }
+
+/**
+ * Extracts the measurement period information from either the options or effective period (in that order depending on presence)
+ * and populates a parameters object including the extracted info to be passed into the parseQueryInfo function
+ */
+export function extractDataRequirementsMeasurementPeriod(options: CalculationOptions, effectivePeriod?: fhir4.Period) {
+  if (!hasMeasurementPeriodInfo(options, effectivePeriod)) {
+    return {};
+  }
+  const parameters: Record<string, Interval> = {};
+
+  if (options.measurementPeriodStart || options.measurementPeriodEnd) {
+    parameters['Measurement Period'] = createIntervalFromEndpoints(
+      options.measurementPeriodStart,
+      options.measurementPeriodEnd
+    );
+  } else {
+    parameters['Measurement Period'] = createIntervalFromEndpoints(effectivePeriod?.start, effectivePeriod?.end);
+  }
+  return parameters;
+}
+
+/**
+ * Creates a cql-execution interval from start to end. If either start or end is not present,
+ * creates an interval with duration exactly one year using the present endpoint
+ */
+export function createIntervalFromEndpoints(start?: string, end?: string) {
+  let startCql, endCql;
+  if (start && end) {
+    ({ startCql, endCql } = Execution.getCQLIntervalEndpoints({
+      measurementPeriodStart: start,
+      measurementPeriodEnd: end
+    }));
+  } else {
+    if (start) {
+      startCql = parseTimeStringAsUTC(start);
+      endCql = new Date(startCql);
+      endCql.setFullYear(startCql.getFullYear() + 1);
+    } else if (end) {
+      endCql = parseTimeStringAsUTC(end);
+      startCql = new Date(endCql);
+      startCql.setFullYear(endCql.getFullYear() - 1);
+    }
+    startCql = DateTime.fromJSDate(startCql, 0);
+    endCql = DateTime.fromJSDate(endCql, 0);
+  }
+  return new Interval(startCql, endCql);
+}
+
+function hasMeasurementPeriodInfo(options: CalculationOptions, effectivePeriod?: fhir4.Period) {
+  return (
+    options.measurementPeriodStart || options.measurementPeriodEnd || effectivePeriod?.start || effectivePeriod?.end
```

Comment:
I have no problem with this being a separate function but do we think it's necessary? Looks like we only use it in one spot and the only reason really why I am asking is it looks like this Branch is not covered through GitHub Actions and adding a test unit for it might just be unnecessary.

Reply:
Yeah I really just did this for readability since its kinda an ugly boolean statement that ultimately accomplishes something really simple. I could convert to const, or if you think it's more readable to just insert the boolean into the function that calls this, happy to do that also :)

Reply:
Gotcha yeah I agree I mostly just pointed this out for branch coverage awareness but not a huge deal at all! Perfect as is as long as branch coverage isn't a concern